File: WeLend - Copy/WeLend Front/src/app/entities/transaction.ts
export interface Transaction {
  transactionID:number;
  reciever:number;
  transactionType:String;
  transactionStatus:String;
  transactionMethod:String;
  transactionPurpose:String;
  exchangeRate:number;
  amount:number;
  currency:String;
  fees:number;
  transactionDate:Date;
  /*@ManyToOne transactionCategory transactionCategory;
  @ManyToOne Account account;*/

}

File: WeLend - Copy/src/main/java/com/pidev/welend/entities/Transaction.java
package com.pidev.welend.entities;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Getter
@Setter
@Entity
@Table( name = "Transaction")
public class Transaction implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="transactionID")
    private Integer transactionID;
    @ManyToOne Account account;
    private Integer reciever;
    @Enumerated(EnumType.STRING)
    private transactionType transactionType;
    @Enumerated(EnumType.STRING)
    private transactionStatus transactionStatus;
    @Enumerated(EnumType.STRING)
    private  transactionMethod transactionMethod;
    private String transactionPurpose;
    private float exchangeRate;
    private float amount;
    private String currency;
    private float fees;
    private Date transactionDate;
    @ManyToOne transactionCategory transactionCategory;



}
